test/integration/model.js

```diff
@@ -517,6 +517,33 @@ module.exports = function(bookshelf) {
         });
       });
 
+      it('ensure events are async', function(ok) {
+          var m = new Site({name: 'new'});
+
+          m.on('saving', function(model) {
+            return new Promise(function (resolve) {
+              setTimeout(function () {
+                model.set('x', 'y');
+                resolve();
+              }, 200);
+            });
+          });
+
+          m.on('saving', function(model) {
+            equal(model.get('x'), 'y');
+            model.unset('x');
+            m.off();
+          });
+
+          m.save(null, {method: 'insert'})
```

Comment:
Do you mind returning the call to `save()` here instead of calling `ok()` at the end? This ensures better consistency with the rest of the test suite and if the test fails the error will get caught correctly instead of being handled by the generic global `onPossiblyUnhandledRejection()` handler.

Reply:
will update 👍